datepicker/operations/index.js

```diff
@@ -0,0 +1,205 @@
+};
+
+Datepicker.prototype._selectDate = function(momentDate, closeOnSelect) {
+  if (this._isOutOfRange(momentDate)) return;
+
+  var selectedDate = this.model.get('selectedDate');
+
+  if (!momentDate.isSame(selectedDate, 'day')) {
+    this._animateText('date', selectedDate);
+    if (!momentDate.isSame(selectedDate, 'year')) this._animateText('year', selectedDate);
+  }
+
+  this.model.set('selectedDate', momentDate);
+  this.emit('selected', this.model.get('selectedDate').format());
+
+  if (closeOnSelect) {
+    this.setValue();
+    this._hide();
+  }
+};
+
+Datepicker.prototype.action = function(action) {
+  this.emit('action', action.text || action);
+
+  if (action.disableDismiss) return;
+
+  this._hide();
+};
+
+Datepicker.prototype._selectYear = function(year) {
+  var momentDate = this.model.get('selectedDate').clone();
+  momentDate.year(year);
+
+  this._selectDate(momentDate);
+  this._setCurrentDate(momentDate);
+  
+  // REVIEW: center on selection or only on view load?
+  this._scrollToSelected(this.$yearlist);
+  this._maybeHideList();
+};
+
+Datepicker.prototype._selectMonth = function(monthIndex) {
+  var date = this.model.get('currentDate').clone();
+  date.month(monthIndex);
+
+  this._setCurrentDate(date);  
+  // REVIEW: center on selection or only on view load?
+  this._scrollToSelected(this.$monthlist);
+  this._maybeHideList();
+};
+
+Datepicker.prototype._maybeHideList = function() {
+  var options = this.getAttribute('options');
+
+  if (!options || !options.disableAutoCloseLists) {
+    this._showView('day');
+    this._flipMonth('right', this._getMonth(this.model.get('currentDate')));
+  }
+};
+
+Datepicker.prototype._animateText = function(text, date) {
+  this.model.set('leaving.' + text, date);
+  this.model.set('animating.' + text, true);
+  // Force style recalculation for transitions to work. See http://stackoverflow.com/questions/18564942/clean-way-to-programmatically-use-css-transitions-from-js/31862081
+  window.getComputedStyle(this.$datepicker).top;
+  this.model.del('animating.' + text);
+};
+
+Datepicker.prototype._scrollToSelected = function($list) {
+  var $item = $list.querySelector('.pickerlist---item--selected');
+
+  if (!$item) return;
+
+  $list.scrollTop = ($item.offsetTop) - (($list.clientHeight / 2) - ($item.clientHeight / 2));
+};
+
+Datepicker.prototype._flipMonth = function(direction, newMonth) {
+  var currentDate = this.model.get('currentDate').clone();
+  this._animateText('month', currentDate);
+
+  // Figure out which view is entering/leaving.
+  var isFirstActive = this.model.set('isMonthViewOne', !this.model.get('isMonthViewOne'));
+  var entering = (isFirstActive) ? 'monthTwo' : 'monthOne';
+  var leaving = (!isFirstActive) ? 'monthTwo' : 'monthOne';
+  var to = (direction === 'right') ? 'left' : 'right';
+
+  // Add next/prev month to entering view.
+  console.log(this.$months.childNodes[0].offsetHeight, this.$months.childNodes[1].offsetHeight);
```

Comment:
Should probably not have console.logging when going into production - aka these should probably be removed.